### PR TITLE
[PAY-1106] Unmount reaction popup animations

### PR DESCRIPTION
**Pull request by dharit-tan**
### Description
Unmount animations for reactions popup (from convos with @dylanjeffers). Using a callback passed to `Animated.timing.start()` that sets the `shouldShowPopup` state to false when the animation completes.

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

Local ios stage

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Review by rickyrombo — APPROVED** [review]

File: packages/mobile/src/screens/chat-screen/ReactionPopup.tsx [hunk 1]
```diff
@@ -95,24 +99,47 @@ export const ReactionPopup = ({
     (r) => r.user_id === userIdEncoded
   )?.reaction
 
-  const beginAnimation = useCallback(() => {
-    Animated.spring(backgroundOpacityAnim.current, {
+  const beginMountAnimation = useCallback(() => {
+    Animated.timing(backgroundOpacityAnim.current, {
```
Comment (dylanjeffers) [inline packages/mobile/src/screens/chat-screen/ReactionPopup.tsx:103]: Now that I see this, can you use the Animated.parallel composer function to just have one .start()?

File: packages/mobile/src/screens/chat-screen/ReactionPopup.tsx [hunk 2]
```diff
@@ -95,24 +99,47 @@ export const ReactionPopup = ({
...
+    }).start()
+    Animated.timing(otherOpacity.current, {
+      toValue: 0,
+      useNativeDriver: true,
+      duration: UNMOUNT_ANIMATION_DURATION_MS
+    }).start(onClose)
+    Animated.spring(translationAnim.current, {
+      toValue: REACTION_CONTAINER_HEIGHT,
+      useNativeDriver: true
+    }).start()
```
Comment (dylanjeffers) [inline packages/mobile/src/screens/chat-screen/ReactionPopup.tsx:133]: And same thing here, can have the callback on the start for the .parallel part

**Review by dylanjeffers — APPROVED** [review]
Awesome changes, lmk if the parallel changes make sense

File: packages/mobile/src/screens/chat-screen/ReactionPopup.tsx (hunk 1 above)
Comment (piazzatron) [inline packages/mobile/src/screens/chat-screen/ReactionPopup.tsx:103]: Support using parallel, I think we could also stand to pull these out (plus the refs) into their own standalone hook to keep the components simpler

**Review by piazzatron — APPROVED** [review]
Looks great!

File: packages/mobile/src/screens/chat-screen/ReactionPopup.tsx (hunk 2 above)
Comment (dharit-tan) [reply packages/mobile/src/screens/chat-screen/ReactionPopup.tsx]: Had to keep the translation animation separate bc it takes long to complete, and while it's incomplete the user wouldn't be able to open a new reaction popup. Couldn't find a way to cut off the animation time or change the duration (not sure we would want to do that anyway) so this seemed like the best solution.

**Pull request merged**